### PR TITLE
Integrate `dirty` state, API requests, and publish button

### PR DESCRIPTION
**Pull request by nylen**
This simple PR integrates post saving/updating introduced in #594 and dirty state introduced in #610.

### Make `SavedState` aware of API requests

**To test:** Edit a post until it shows "Unsaved changes" in the top left.  Save it via the Update/Save draft button and observe that the text changes from "Unsaved changes" to "Saved".

### Make `PublishButton` aware of `dirty` state

**To test:** Save a post using the Update/Save draft button.  The button will show "Updated!" or "Saved!". 
 Edit a post until it shows "Unsaved changes" in the top left.  Verify that the button text changes back to "Update" or "Save".

### Next steps

In short: **error handling**.  Currently errors are indicated by the `PublishButton` showing "Update failed" or "Save failed".  In this PR, the error state is **cleared** when the post is modified again.  However, really we probably want a red error icon/message next to the publish button instead, and this would stick around after the post is edited.  This should be a separate task that builds on @jasmussen's mockups from https://github.com/WordPress/gutenberg/pull/594#issuecomment-298871748.

### CODE REVIEW

**Review by aduth — APPROVED** [review]
Looks good 👍
